### DISCUSSION TITLE
using usrsctp lib to send STCP packet in dtls encapsulation

### DISCUSSION THREAD

**Discussion by bigsource** [sctplab/usrsctp#682] [Q&A]
I want to encapsulation STCP packet in DTLS(Datagram Transport Layer Security) before sending from UDP,  the encapsulation layers is "appdata/SCTP/DTLS/UDP/IP".

from openssl related API, dtls need to configured via socket fd, but usrsctp lib API has not provide socket fd information.
is it convinent to encapsulation STCP in DTLS?

**Comment by tuexen** [comment]
The API is intended to provide this. Have a look at the examples in the `programs` directory starting with `ekr`, for example `ekr_loop.c`:
- For the sending side, implement `conn_output` and pass the SCTP packet to a DTLS `send` call.
- For the receiving side, get packets from the DTLS library and pass them to `usrsctp_conninput()`.
